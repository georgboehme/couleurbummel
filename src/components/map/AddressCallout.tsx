import {makeStyles} from '@rneui/themed';
import {useTheme} from '@rneui/themed';
import React, {memo, useContext} from 'react';
import {ColorValue, Text, View} from 'react-native';

import {Colours, MapDisplayables} from '../../types/model';
import ColoursView from '../Colours';
import {ModelContext} from '../contexts/ModelContext';
import {OrganisationIndicator} from '../OrganisationIndicator';

type AddressCalloutProps = {
  title: string;
  distanceString?: string;
  mapDisplayables: MapDisplayables;
};

interface CalloutCellProps {
  id: string;
  colours?: Colours[];
  shortName: string;
  indicatorColourId?: string;
}

/**
 * The address callout shown when a marker is tapped
 */
export const AddressCallout = memo(
  ({title, distanceString, mapDisplayables}: AddressCalloutProps) => {
    const styles = useStyles();
    const model = useContext(ModelContext);

    return (
      <View style={styles.container}>
        <View style={styles.headerRow}>
          <Text numberOfLines={1} style={styles.calloutHeaderText}>
            {title}
          </Text>
          <View style={styles.calloutDistance}>
            <Text numberOfLines={1} style={styles.calloutDistanceText}>
              {distanceString}
            </Text>
          </View>
        </View>
        <View>
          {mapDisplayables.corporations.map(c => (
            <CalloutCell
              key={c.id}
              id={c.id}
              colours={c.colours}
              shortName={c.shortName}
              indicatorColourId={c.indicatorColourId}
            />
          ))}
          {mapDisplayables.pois.map(p => {
            const colours = model.corporation(
              p.associatedCorporationIds?.[0],
            )?.colours;
            return (
              <CalloutCell
                key={p.id}
                id={p.id}
                shortName={p.shortName}
                indicatorColourId={p.indicatorColourId}
                {...(colours && {colours})}
              />
            );
          })}
        </View>
      </View>
    );
  },
);

const CalloutCell = ({
  id,
  shortName,
  indicatorColourId,
  colours,
}: CalloutCellProps) => {
  const {theme} = useTheme();
  const model = useContext(ModelContext);
  const styles = useStyles();

  const indicatorColour = model.colour(indicatorColourId) || theme.colors.grey0;

  const cols = (colours?.[0]?.colourIds || [])
    .map(c => model.colour(c))
    .filter(c => !!c)
    .map(c => c as ColorValue);

  return (
    <View style={styles.cell}>
      <OrganisationIndicator colour={indicatorColour} />
      <Text style={styles.cellName}>{shortName}</Text>
      {colours && (
        <View style={styles.cellColours}>
          <ColoursView
            key={`colours_${id}`}
            coloursKey={`colours_${id}`}
            colours={cols}
            percussion={model.colour(colours?.[0].percussionColourId)}
            baseColour={model.colour(colours?.[0].baseColourId)}
          />
        </View>
      )}
    </View>
  );
};

const useStyles = makeStyles(theme => ({
  container: {
    width: '100%',
    backgroundColor: 'transparent',
  },
  headerRow: {
    flexDirection: 'row',
  },
  calloutHeaderText: {
    textAlign: 'auto',
    fontWeight: 'bold',
    fontSize: 18,
    marginBottom: 5,
    flex: 1,
    color: theme.colors.fixed.black,
  },
  calloutDistance: {
    marginBottom: 5,
    justifyContent: 'center',
    minWidth: '20%',
  },
  calloutDistanceText: {
    color: theme.colors.fixed.darkGrey,
    fontSize: 11,
    textAlign: 'center',
  },
  cell: {
    flexDirection: 'row',
    flex: 1,
    alignItems: 'center',
    justifyContent: 'space-between',
    padding: 5,
  },
  cellName: {
    flexWrap: 'nowrap',
    flex: 1,
    fontSize: 18,
    marginHorizontal: 5,
    color: theme.colors.fixed.black,
  },
  cellColours: {
    alignSelf: 'center',
    width: '20%',
  },
}));
