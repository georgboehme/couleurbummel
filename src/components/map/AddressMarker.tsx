import {Image, makeStyles, useTheme} from '@rneui/themed';
import React, {memo, useContext, useMemo} from 'react';
import {ColorValue, StyleSheet, View} from 'react-native';
import {Point} from 'react-native-maps/lib/sharedTypes';
import Svg, {Circle, G, Path} from 'react-native-svg';

import getIconNamed from '../../assets/icons';
import {MapDisplayables} from '../../types/model';
import {ModelContext} from '../contexts/ModelContext';

interface AddressMarkerProps {
  mapDisplayables: MapDisplayables;
}

/**
 * The marker shown on the map
 */
export const AddressMarker = memo(({mapDisplayables}: AddressMarkerProps) => {
  const model = useContext(ModelContext);

  const colours = useMemo(
    () =>
      model.colourValues(
        [...mapDisplayables.corporations, ...mapDisplayables.pois]
          .map(c => c.indicatorColourId)
          .concat(mapDisplayables.pois.map(c => c.indicatorColourId))
          .filter(c => !!c)
          .map(c => c as string),
      ),
    [model, mapDisplayables],
  );

  /* If we have only POIs, we show their icon on a neutral background */
  if (
    !mapDisplayables.corporations.length &&
    mapDisplayables.pois.length &&
    mapDisplayables.pois[0].iconName
  ) {
    return <AddressMarkerView colours={[]} iconName={'icon_coat_mch'} />;
  }

  return <AddressMarkerView colours={colours} />;
});

const getCoordinatesForPercent = (
  percent: number,
  radius: number,
  centre: Point,
) => {
  const angle = 2 * Math.PI * percent;
  const target = {
    x: centre.x + radius * Math.cos(angle),
    y: centre.y + radius * Math.sin(angle),
  };

  return target;
};

/**
 * Returns the paths to draw the slices of a pie chart in the given colours
 *
 * For each slice, we move to the start point
 * @param colours
 * @param radius
 * @param centre
 */
const pathForColours = (
  colours: ColorValue[],
  radius: number,
  centre: Point,
) => {
  let cumulativePercentage = 0;

  // Start at the top of the circle
  let previous = getCoordinatesForPercent(0, radius, centre);

  return colours.map((col, i) => {
    cumulativePercentage += 1 / colours.length;

    const target = getCoordinatesForPercent(
      cumulativePercentage,
      radius,
      centre,
    );

    // We're drawing each wedge individually and more than one. So we always want to draw the short arc.
    const largeArcFlag = 0;

    // Draw the wedge
    const pathData = [
      `M ${previous.x} ${previous.y}`, // Move to start
      `A ${radius} ${radius} 0 ${largeArcFlag} 1 ${target.x} ${target.y}`, // Arc to the next coordinate
      `L ${centre.x} ${centre.y}`, // Line to centre
    ].join(' ');

    previous = target;

    return (
      <Path
        transform={`rotate(-90, ${centre.x}, ${centre.y})`}
        key={`slice_${i}`}
        d={pathData}
        fill={col}
      />
    );
  });
};

export const AddressMarkerView = memo(
  ({colours, iconName}: {colours: ColorValue[]; iconName?: string}) => {
    const {theme} = useTheme();
    const styles = useStyles();

    const centreStrokeColour = colours.length
      ? theme.colors.fixed.black
      : theme.colors.mapMarkerFill;

    let centreFillColour: ColorValue = 'transparent';
    if (colours.length === 0 && iconName) {
      centreFillColour = theme.colors.mapMarkerFill;
    }
    if (colours.length === 1) {
      centreFillColour = colours[0];
    }

    colours.sort();

    return (
      <View>
        <View style={styles.pin}>
          <Svg height="100%" width="100%" viewBox="0 0 210 297">
            <G transform="translate(129.53117,73.448518)">
              <Path
                d={
                  'm -25.025404,-67.058727 c -45.32449,0 -82.066306,36.742196 -82.066306,82.065275 0,4.763553 0.82687,10.552168 2.30218,17.093013 9.699623,11.693867 22.660072,25.824516 38.471556,39.712305 21.739555,19.094783 44.542681,34.003844 67.775191,44.313054 6.3887991,2.83493 12.816193,5.31994 19.272725,7.45743 18.651695,-40.130906 37.299533,-85.91832 37.299533,-108.575802 0,-45.323079 -36.742171,-82.065275 -82.066309,-82.065275 -0.06773,0 -0.494027,0.0021 -0.494027,0.0021 0,0 -0.427162,-0.0021 -0.494543,-0.0021 z m 0.116272,41.017134 c 0.05151,0 0.378271,0.0047 0.378271,0.0047 0,0 0.326413,-0.0047 0.378272,-0.0047 21.0502278,0 38.230224,17.0657417 38.230224,38.118087 0,21.051992 -17.2352894,38.118087 -38.285518,38.118087 -0.05186,0 -0.322978,-0.0041 -0.322978,-0.0041 0,0 -0.271988,0.0041 -0.323494,0.0041 -21.050581,0 -38.285519,-17.066095 -38.285519,-38.118087 0,-21.0523453 17.180161,-38.118087 38.230742,-38.118087 z m -71.18687,86.949732 c 2.888542,8.085306 6.206886,16.581123 9.777698,25.214998 3.238497,3.092094 6.622888,6.201435 10.149252,9.299174 21.739555,19.094429 44.542165,34.002819 67.774674,44.312029 5.9905133,2.65817 12.0154312,5.00568 18.065564,7.05073 1.837971,-3.76837 3.715013,-7.65704 5.611544,-11.63495 C 8.8590062,132.97207 2.4655418,130.47282 -3.8897434,127.64567 -28.330868,116.77413 -52.247642,101.1174 -74.975327,81.11104 -82.709972,74.302789 -89.759061,67.479324 -96.096002,60.908139 Z m 21.681364,52.605061 c 20.065274,44.27215 41.946794,84.70439 41.946794,84.70439 2.09832,3.87525 5.707782,5.75042 7.936983,5.6534 2.229201,0.097 5.838311,-1.77815 7.936983,-5.6534 0,0 8.9257808,-16.49329 20.6023605,-39.94382 -5.945005,-2.06163 -11.8637297,-4.39986 -17.7498205,-7.01817 -20.737316,-9.22443 -41.096623,-21.89493 -60.6733,-37.7424 z'
                }
                stroke={theme.colors.fixed.black}
                strokeWidth={2}
                fill={theme.colors.mapMarkerFill}
              />
            </G>
            {
              // If there's more than one colour, we'll draw a pie chart
              colours.length > 1 &&
                pathForColours(colours, 38.5, {x: 105.60272, y: 85.525055})
            }
            <Circle
              cx="105.60272"
              cy="85.525055"
              r="38.75"
              fill={centreFillColour}
              stroke={centreStrokeColour}
              strokeWidth={2}
            />
          </Svg>

          {iconName && (
            <View style={styles.iconContainer}>
              <Image style={styles.icon} source={getIconNamed(iconName)} />
            </View>
          )}
        </View>
      </View>
    );
  },
);

const useStyles = makeStyles(() => ({
  pin: {
    alignSelf: 'center',
    width: 40,
    height: 67,
  },
  iconContainer: {
    ...StyleSheet.absoluteFillObject,
    flexDirection: 'row',
    justifyContent: 'center',
    top: 10,
  },
  icon: {
    width: 25,
    height: 25,
  },
}));
