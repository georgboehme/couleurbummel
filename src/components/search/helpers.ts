import log from '../../log';
import {CouleurbummelModel} from '../../model/model';
import {Address, Colours, Corporation} from '../../types/model';

export interface ColoursQuery {
  baseColourId: string;
  colourFamilyIds: string[];
}

const coloursMatch = (
  colours: Colours,
  colourIdsToFind: string[][],
  baseColourIdsToFind?: string[],
) => {
  // If we're searching for more or fewer colours than we have, it's not a match
  if (colourIdsToFind.length !== colours.colourIds.length) {
    return false;
  }

  // If the query contains a base colour, but the corporation doesn't have one or has the wrong one, it doesn't match.
  if (
    baseColourIdsToFind?.length &&
    (!colours.baseColourId ||
      baseColourIdsToFind.indexOf(colours.baseColourId) === -1) &&
    (!colours.percussionColourId ||
      baseColourIdsToFind.indexOf(colours.percussionColourId) === -1)
  ) {
    log.debug(
      'Base colour not matching',
      colours.baseColourId,
      baseColourIdsToFind,
    );
    return false;
  }

  let index = 0;
  let coloursFound = 0;

  // See if we can find the colours in the correct order.
  for (const i in colourIdsToFind) {
    const matchingColourIds = colourIdsToFind[i];

    if (
      // Blank colours are wildcards and always match
      !matchingColourIds.length ||
      matchingColourIds.indexOf(colours.colourIds[index]) !== -1
    ) {
      coloursFound++;
    }
    index++;
  }

  return coloursFound === colourIdsToFind.length;
};

export const searchColours = (
  query: ColoursQuery,
  data: [Address, Corporation][],
  model: CouleurbummelModel,
) => {
  log.debug('Serving query', query);

  const baseColourToSearch = model.colourFamily(
    query.baseColourId,
  )?.memberColourIds;

  const coloursToSearch = (query.colourFamilyIds || []).map(
    familyId => model.colourFamily(familyId)?.memberColourIds || [],
  );

  log.debug('Resolved colours', baseColourToSearch, coloursToSearch);

  return data.filter(([_address, corporation]) => {
    return (corporation.colours || []).some(cols =>
      coloursMatch(cols, coloursToSearch, baseColourToSearch),
    );
  });
};
