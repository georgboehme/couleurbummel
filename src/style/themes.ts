import {DarkTheme, DefaultTheme} from '@react-navigation/native';
import {Theme} from '@react-navigation/native/lib/typescript/src/types';
import {createTheme, darkColors, lightColors} from '@rneui/themed';
import {ColorValue, Platform} from 'react-native';

const ColourConstants = {
  primary: '#008040',
  primaryTransparent: 'rgba(0,128,64,0.75)',
  primaryDark: '#006633',
  accent: '#3CB371',
  accentTransparent: '#3CB37180',
  white: '#FFFFFF',
  whiteDarkModeText: '#E1E8EE',
  grey: '#EBEBEB',
  darkGrey: '#393E42FF',
  black: '#000000',
};

const sharedColours = {
  primary: ColourConstants.primary,
  primaryTransparent: ColourConstants.primaryTransparent,
  accent: ColourConstants.accent,
  accentTransparent: ColourConstants.accentTransparent,
  mapMarkerFill: ColourConstants.grey,
  mapMarkerCentreStroke: ColourConstants.black,
  iconOnPrimaryColourBackgroundColour: ColourConstants.whiteDarkModeText,
  fixed: {
    white: ColourConstants.white,
    lightGrey: ColourConstants.grey,
    darkGrey: ColourConstants.darkGrey,
    black: ColourConstants.black,
  },
};

const platformLightDefaults = Platform.select({
  default: lightColors.platform.android,
  ios: lightColors.platform.ios,
});

const platformDarkDefaults = Platform.select({
  default: darkColors.platform.android,
  ios: darkColors.platform.ios,
});

export const theme = createTheme({
  lightColors: {
    ...platformLightDefaults,
    ...sharedColours,
    text: ColourConstants.black,
  },
  darkColors: {
    ...platformDarkDefaults,
    ...sharedColours,
    text: ColourConstants.whiteDarkModeText,
  },
  components: {
    Icon: {type: 'material-community', color: ColourConstants.primary},
    ListItem: {
      containerStyle: {
        // Required to allow the selection highlight (defaults to theme background otherwise)
        backgroundColor: 'transparent',
      },
    },
    // Compiler warning fixed in https://github.com/react-native-elements/react-native-elements/pull/3691
    // @ts-ignore
    ListItemSwipeable: {
      containerStyle: {
        // Required to allow the selection highlight (defaults to theme background otherwise)
        backgroundColor: 'transparent',
      },
    },
    SearchBar: {
      containerStyle: {
        backgroundColor: 'transparent',
        borderTopColor: 'transparent',
        borderBottomColor: 'transparent',
      },
    },
  },
});

export const navigationTheme: {lightTheme: Theme; darkTheme: Theme} = {
  lightTheme: {
    dark: false,
    colors: {
      ...DefaultTheme.colors,
      primary: sharedColours.primary,
    },
  },
  darkTheme: {
    dark: true,
    colors: {
      ...DarkTheme.colors,
      primary: sharedColours.primary,
    },
  },
};

declare module '@rneui/themed' {
  export interface Colors {
    primaryTransparent: ColorValue;
    accent: ColorValue;
    accentTransparent: ColorValue;
    mapMarkerFill: ColorValue;
    mapMarkerCentreStroke: ColorValue;
    text: ColorValue;
    iconOnPrimaryColourBackgroundColour: ColorValue;
    fixed: {
      white: ColorValue;
      lightGrey: ColorValue;
      darkGrey: ColorValue;
      black: ColorValue;
    };
  }
}
