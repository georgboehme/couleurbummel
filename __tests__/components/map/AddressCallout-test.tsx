/**
 * @format
 */

import 'react-native';

import {ThemeProvider} from '@rneui/themed';
import {render, screen, waitFor} from '@testing-library/react-native';
import React from 'react';

// Note: test renderer must be required after react-native.
import {AddressCallout} from '../../../src/components/map/AddressCallout';
import {theme} from '../../../src/style/themes';
import {getTestModel} from '../../__helpers__/helpers';

describe('AddressCallout', () => {
  const model = getTestModel();

  it('renders multiple corporations', async () => {
    const corporationsAtSameAddress = model.corporationsById([
      'corporation_de_aachen_adv_laetitia',
      'corporation_de_aachen_kstv_alaniabreslau',
    ]);

    expect(corporationsAtSameAddress).toHaveLength(2);

    render(
      <ThemeProvider theme={theme}>
        <AddressCallout
          title={'Street or house name'}
          distanceString={'42m'}
          mapDisplayables={{
            corporations: corporationsAtSameAddress,
            pois: [],
          }}
        />
      </ThemeProvider>,
    );

    await waitFor(() => expect(screen.toJSON()).toMatchSnapshot());
  });

  it('renders a POI', async () => {
    const poi = model.pointOfInterest('poi_aachen_townhall');
    expect(poi).toBeTruthy();

    render(
      <ThemeProvider theme={theme}>
        <AddressCallout
          title={'Street or house name'}
          distanceString={'42m'}
          mapDisplayables={{
            corporations: [],
            pois: [poi!],
          }}
        />
      </ThemeProvider>,
    );

    await waitFor(() => expect(screen.toJSON()).toMatchSnapshot());
  });
});
