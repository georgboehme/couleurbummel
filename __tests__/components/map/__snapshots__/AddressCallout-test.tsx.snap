// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddressCallout renders a POI 1`] = `
<View
  style={
    {
      "backgroundColor": "transparent",
      "width": "100%",
    }
  }
>
  <View
    style={
      {
        "flexDirection": "row",
      }
    }
  >
    <Text
      numberOfLines={1}
      style={
        {
          "color": "#000000",
          "flex": 1,
          "fontSize": 18,
          "fontWeight": "bold",
          "marginBottom": 5,
          "textAlign": "auto",
        }
      }
    >
      Street or house name
    </Text>
    <View
      style={
        {
          "justifyContent": "center",
          "marginBottom": 5,
          "minWidth": "20%",
        }
      }
    >
      <Text
        numberOfLines={1}
        style={
          {
            "color": "#393E42FF",
            "fontSize": 11,
            "textAlign": "center",
          }
        }
      >
        42m
      </Text>
    </View>
  </View>
  <View>
    <View
      style={
        {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "space-between",
          "padding": 5,
        }
      }
    >
      <View
        style={
          {
            "aspectRatio": 1,
            "height": 20,
            "marginRight": 5,
          }
        }
      >
        <View
          style={{}}
        >
          <RNSVGSvgView
            align="xMidYMid"
            bbHeight="100%"
            bbWidth="100%"
            focusable={false}
            height="100%"
            meetOrSlice={0}
            minX={0}
            minY={0}
            style={
              [
                {
                  "backgroundColor": "transparent",
                  "borderWidth": 0,
                },
                {
                  "flex": 0,
                  "height": "100%",
                  "width": "100%",
                },
              ]
            }
            vbHeight={100}
            vbWidth={100}
            width="100%"
          >
            <RNSVGGroup>
              <RNSVGEllipse
                cx="50%"
                cy="50%"
                fill={
                  {
                    "payload": 4281941570,
                    "type": 0,
                  }
                }
                propList={
                  [
                    "fill",
                    "stroke",
                    "strokeWidth",
                  ]
                }
                rx="50%"
                ry="50%"
                stroke={
                  {
                    "payload": 4290493371,
                    "type": 0,
                  }
                }
                strokeWidth="2"
              />
            </RNSVGGroup>
          </RNSVGSvgView>
        </View>
      </View>
      <Text
        style={
          {
            "color": "#000000",
            "flex": 1,
            "flexWrap": "nowrap",
            "fontSize": 18,
            "marginHorizontal": 5,
          }
        }
      >
        Rathaus
      </Text>
    </View>
  </View>
</View>
`;

exports[`AddressCallout renders multiple corporations 1`] = `
<View
  style={
    {
      "backgroundColor": "transparent",
      "width": "100%",
    }
  }
>
  <View
    style={
      {
        "flexDirection": "row",
      }
    }
  >
    <Text
      numberOfLines={1}
      style={
        {
          "color": "#000000",
          "flex": 1,
          "fontSize": 18,
          "fontWeight": "bold",
          "marginBottom": 5,
          "textAlign": "auto",
        }
      }
    >
      Street or house name
    </Text>
    <View
      style={
        {
          "justifyContent": "center",
          "marginBottom": 5,
          "minWidth": "20%",
        }
      }
    >
      <Text
        numberOfLines={1}
        style={
          {
            "color": "#393E42FF",
            "fontSize": 11,
            "textAlign": "center",
          }
        }
      >
        42m
      </Text>
    </View>
  </View>
  <View>
    <View
      style={
        {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "space-between",
          "padding": 5,
        }
      }
    >
      <View
        style={
          {
            "aspectRatio": 1,
            "height": 20,
            "marginRight": 5,
          }
        }
      >
        <View
          style={{}}
        >
          <RNSVGSvgView
            align="xMidYMid"
            bbHeight="100%"
            bbWidth="100%"
            focusable={false}
            height="100%"
            meetOrSlice={0}
            minX={0}
            minY={0}
            style={
              [
                {
                  "backgroundColor": "transparent",
                  "borderWidth": 0,
                },
                {
                  "flex": 0,
                  "height": "100%",
                  "width": "100%",
                },
              ]
            }
            vbHeight={100}
            vbWidth={100}
            width="100%"
          >
            <RNSVGGroup>
              <RNSVGEllipse
                cx="50%"
                cy="50%"
                fill={
                  {
                    "payload": 4281941570,
                    "type": 0,
                  }
                }
                propList={
                  [
                    "fill",
                    "stroke",
                    "strokeWidth",
                  ]
                }
                rx="50%"
                ry="50%"
                stroke={
                  {
                    "payload": 4290493371,
                    "type": 0,
                  }
                }
                strokeWidth="2"
              />
            </RNSVGGroup>
          </RNSVGSvgView>
        </View>
      </View>
      <Text
        style={
          {
            "color": "#000000",
            "flex": 1,
            "flexWrap": "nowrap",
            "fontSize": 18,
            "marginHorizontal": 5,
          }
        }
      >
        A.D.V. Laetitia
      </Text>
      <View
        style={
          {
            "alignSelf": "center",
            "width": "20%",
          }
        }
      >
        <View
          style={
            {
              "aspectRatio": 1.5,
            }
          }
        >
          <View
            style={
              [
                {
                  "borderColor": "#bbb",
                  "borderRadius": 5,
                  "borderWidth": 1,
                  "flexGrow": 1,
                  "flexShrink": 1,
                  "justifyContent": "center",
                  "overflow": "hidden",
                  "width": "100%",
                },
              ]
            }
          />
        </View>
      </View>
    </View>
    <View
      style={
        {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "space-between",
          "padding": 5,
        }
      }
    >
      <View
        style={
          {
            "aspectRatio": 1,
            "height": 20,
            "marginRight": 5,
          }
        }
      >
        <View
          style={{}}
        >
          <RNSVGSvgView
            align="xMidYMid"
            bbHeight="100%"
            bbWidth="100%"
            focusable={false}
            height="100%"
            meetOrSlice={0}
            minX={0}
            minY={0}
            style={
              [
                {
                  "backgroundColor": "transparent",
                  "borderWidth": 0,
                },
                {
                  "flex": 0,
                  "height": "100%",
                  "width": "100%",
                },
              ]
            }
            vbHeight={100}
            vbWidth={100}
            width="100%"
          >
            <RNSVGGroup>
              <RNSVGEllipse
                cx="50%"
                cy="50%"
                fill={
                  {
                    "payload": 4281941570,
                    "type": 0,
                  }
                }
                propList={
                  [
                    "fill",
                    "stroke",
                    "strokeWidth",
                  ]
                }
                rx="50%"
                ry="50%"
                stroke={
                  {
                    "payload": 4290493371,
                    "type": 0,
                  }
                }
                strokeWidth="2"
              />
            </RNSVGGroup>
          </RNSVGSvgView>
        </View>
      </View>
      <Text
        style={
          {
            "color": "#000000",
            "flex": 1,
            "flexWrap": "nowrap",
            "fontSize": 18,
            "marginHorizontal": 5,
          }
        }
      >
        K.St.V. Alania-Breslau
      </Text>
      <View
        style={
          {
            "alignSelf": "center",
            "width": "20%",
          }
        }
      >
        <View
          style={
            {
              "aspectRatio": 1.5,
            }
          }
        >
          <View
            style={
              [
                {
                  "borderColor": "#bbb",
                  "borderRadius": 5,
                  "borderWidth": 1,
                  "flexGrow": 1,
                  "flexShrink": 1,
                  "justifyContent": "center",
                  "overflow": "hidden",
                  "width": "100%",
                },
              ]
            }
          />
        </View>
      </View>
    </View>
  </View>
</View>
`;
